### PR TITLE
Replace safe approves

### PR DESCRIPTION
**Pull request by DaigaroCota**
This pull request addresses Item #-6 of the Three Sigma security audit.

### CODE REVIEW

File: packages/protocol/src/routers/ConnextHandler.sol [hunk 1]
```diff
@@ -216,7 +216,7 @@ contract ConnextHandler {
       revert ConnextHandler__executeFailed_tranferAlreadyExecuted(transferId, nonce);
     }
 
-    IERC20(txn.asset).approve(address(connextRouter), txn.amount);
+    SafeERC20.safeIncreaseAllowance(IERC20(txn.asset), address(connextRouter), txn.amount);
 
     try connextRouter.xBundle(actions, args) {
```
Comment (0x73696d616f) [inline packages/protocol/src/routers/ConnextHandler.sol:221]: Think the allowance should be decreased if the `xBundle(...)` call fails (assets would not have been pulled)

**Review by 0x73696d616f — CHANGES_REQUESTED** [review]
Just left 1 note, everything else looks good

File: packages/protocol/src/routers/ConnextHandler.sol (hunk 1 above)
Comment (DaigaroCota) [reply packages/protocol/src/routers/ConnextHandler.sol:221]: Added decrease allowance in catch as suggested, considering failed execution.

**Review by 0x73696d616f — APPROVED** [review]
Looks good

**Review by pedrovalido — APPROVED** [review]

**Pull request merged**